extension/scripts/Selector/SelectorTable.js

```diff
@@ -153,24 +176,74 @@ var SelectorTable = {
         var dataColumns = [];
         this.columns.forEach(function (column) {
             if (column.extract === true) {
-                dataColumns.push(column.name);
+                dataColumns.push(column.name.replace(/^\s*/,'').replace(/\s*$/,''));
             }
         });
         return dataColumns;
     },
 
-    getFeatures: function () {
-        return ['selector', 'multiple', 'columns', 'delay', 'tableDataRowSelector', 'tableHeaderRowSelector', 'tableAddMissingColumns']
-    },
 
-    getItemCSSSelector: function () {
-        return "table";
+
+
+    /**
+     * Extract table header column info from html
+     * @param html
+     */
+    getTableHeaderColumnsFromHTML: function (headerRowSelector, html) {
+        var $table = $(html);
+        var $headerRowColumns = $table.find(headerRowSelector);
+
+        if ($headerRowColumns.length > 1) {
+            $headerRowColumns = $headerRowColumns.find("th:first-child")
+        } else if ($headerRowColumns.find("th").length) {
+            $headerRowColumns = $headerRowColumns.find("th");
+        } else if ($headerRowColumns.find("td").length) {
+            $headerRowColumns = $headerRowColumns.find("td");
+        }
+        verticalTable = $("#edit-selector [name=verticalTable]").is(":checked");
+        var columns = [];
+        if (verticalTable){
+            $headerRowColumns.prevObject.each(function (i, columnEl) {
+                var header = columnEl.innerText;
+                var name = header;
+                if (header.length !== 0) {
+                    columns.push({
+                        header: header,
+                        name: name,
+                        extract: true
+                    });
+                }
+            });
+        } else {
+            $headerRowColumns.each(function (i, columnEl) {
+            var header = columnEl.innerText;
+            var name = header;
+            if (header.length !== 0) {
+                columns.push({
+                    header: header,
+                    name: name,
+                    extract: true
+                });
+            }
+        });
+        }
+        return columns;
     },
 
+
+
+    /**
+     * Рабочие методы
+     */
+
     getTableHeaderRowSelectorFromTableHTML: function (html) {
         var $table = $(html);
         var firstRow = $table.find("tr:first-child");
         var rowCount = $table.find("tr").length;
+        var verticalTable = this.verticalTable;
+        if (this.verticalTable === undefined) {
+            verticalTable = $("#edit-selector [name=verticalTable]").is(":checked");
+        }
         if ($table.find("thead tr:has(td:not(:empty)), thead tr:has(th:not(:empty))").length) {
```

Comment:
So many nested if expressions is hard to understand. Whe should discuss how to refactor it.